### DISCUSSION TITLE
No module named 'sktime.utils.data_container' is the error

### DISCUSSION THREAD

**Discussion by monasound** [alan-turing-institute/sktime#4384] [Q&A]
This is very similar to https://github.com/sktime/sktime/issues/451

I am trying to import similar kind of data structure to that of mentioned in Issue No -451. 

I am getting the following error when trying to follow the solution to the above mentioned issue.

from sktime.utils.data_container import from_2d_array_to_nested
df = pd.read_csv("data_file.csv")
nested = from_2d_array_to_nested(df)

Error: No module named 'sktime.utils.data_container'. 

I am totally new to Python. My aim is to do time-series data classification. Appreciate your help on this issue.

Following is the package versions
Python - 3.8.16
Spyder - 5.3.3.
Sktime - 0.14.1
Sktime-all-extras - 0.14.1
numpy - 1.23.5
pandas - 1.5.3

**Comment by fkiraly** [comment]
oh, that's an ancient issue (I think from before version 0.8.0), and the modules you're trying to import does no longer exist!

I'd also recommend to follow the tutorials on binder:
https://mybinder.org/v2/gh/sktime/sktime/main?filepath=examples

Since a while, there's also no need to convert your data into the nested format anymore, by the way.

On a side note, I would recommend, you should try to upgrade your environment, `sktime` is now on version 0.16.1, pandas is now on version 2, etc.

**Reply by monasound** [reply]
Thank you for your reply. 

Now I have upgraded to Sktime - 0.16.1

I have imported a sample Csv file similar to that of Issue 451. However, when fitting the forest classifier, I am getting error related to its TYPE.  

Erorr: X is not of a supported input data type.X must be in a supported mtype format for Panel, found <class 'pandas.core.frame.DataFrame'>Use datatypes.check_is_mtype to check conformance with specifications.

I appreciate your help again.

Variables: LNUID, T0, T1,T2,T3,T4,T5,T6, Ori_Intv2


import pandas as pd
import sktime

from sktime.datasets import load_from_tsfile_to_dataframe
from sktime.classification.interval_based import TimeSeriesForestClassifier
from sktime.datasets import load_arrow_head
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

df = pd.read_csv('D:\\Spatial Analysis\\SAMPLE.csv')
#convert pandas dataframe to numpy array
##arr = df.to_numpy()
df = df.set_index("LNU_ID")

y = df.pop('Ori_Intv2')
X = df
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.2)

classifier = TimeSeriesForestClassifier()
classifier.fit(X_train, y_train)
y_pred = classifier.predict(X_test)
accuracy_score(y_test, y_pred)

**Reply by monasound** [reply]
Updated: It is working when converted the data frame to 2D and 1D numpy array. Thank you for all your help.

**Reply by monasound** [reply]
@fkiraly 

Could you please confirm that the below multiple time-series classification using SKtime Forest classifier is correct in terms of data input format? 


import pandas as pd
import sktime

from sktime.classification.interval_based import TimeSeriesForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
import numpy as np

import random
from sklearn.metrics import accuracy_score, confusion_matrix, precision_score, recall_score, ConfusionMatrixDisplay

![image](https://user-images.githubusercontent.com/103995157/229166825-3639c9a0-98f6-4bc7-b3b6-246dd225d750.png)


df = pd.read_csv('D:\\RF_inputs\\Sample_Github_TS.csv')

### Shuffle the data
df_shuffled = df.sample(frac=1, random_state=123).reset_index(drop=True)
df_shuffled = df_shuffled.set_index("Land_ID")

###convert pandas dataframe to numpy array
df_numpy = df_shuffled.to_numpy()

###Split the data
train, test = train_test_split(df_numpy , random_state = 123,test_size=0.3, shuffle=True)
X_train =train[:,0:10]
y_train =train[:,10]


X_test =test[:,0:10]
y_test =test[:,10]


### Classifier
classifier = TimeSeriesForestClassifier(random_state=123)
classifier.fit(X_train, y_train)


y_pred = classifier.predict(X_test)
accuracy_score(y_test, y_pred)


### Create the confusion matrix
TSF_test = confusion_matrix(y_pred,y_test)
ConfusionMatrixDisplay(confusion_matrix=TSF_test).plot();


I also have a question on how to retain Land_ID as rowname like in pd dataframe. This was not possible with numpy2D array. I appreciate your reply. Is there any other format that I can convert?

I am referring to this page https://github.com/sktime/sktime/blob/main/examples/AA_datatypes_and_datasets.ipynb

Since I am new to python, it is taking time for me to learn different data formats. Appreciate your help
